docs/TESTPLAN.md

```diff
@@ -57,6 +57,14 @@ modifications/workarounds in `src/test/testRunner.ts`.
     -   [Mocha](https://mochajs.org/) framework is used to write tests.
 -   New code requires new tests.
 
+## Best Practices
```

Comment:
this belongs in CONTRIBUTING.md, rather than TESTPLAN which is only an architectural/strategic doc

Reply:
https://github.com/aws/aws-toolkit-vscode/pull/1892